tests/python_package_test/test_basic.py

```diff
@@ -3,6 +3,7 @@
 
 import lightgbm as lgb
 import numpy as np
+import pandas as pd
```

Comment:
@jmoralez Please make `pandas` optional for basic tests like you did in sklearn tests:
```
pd = pytest.importorskip("pandas")
```

Reply:
Since I had some `pd.Series` in the parametrize for `y` I changed it to this:
```python
    ('y', 'is_series'),
    [
        (np.random.rand(10), False),
        (np.random.rand(10, 1), False),
        ([1] * 10, False),
        (np.random.rand(10), True),
        (['a', 'b'], True),
        ([[1], [2]], False)
    ])
```
and then I do this inside the function
```python
    if is_series:
        pd = pytest.importorskip('pandas')
        y = pd.Series(y)
```
Is that the best way to do it?

Reply:
Maybe it will be easier to just mark the whole test for skipping if `pandas` is not installed?
```
@pytest.mark.skipif(not lgb.compat.PANDAS_INSTALLED, reason='pandas is not installed')
```
like here
https://github.com/microsoft/LightGBM/blob/50e061f396d7c1b704e79371dea0ddf07680ba75/tests/python_package_test/test_plotting.py#L132

Yeah, it will skip the whole test if pandas is not installed but it looks much easier to read and maintain than one more parametrization argument.

Reply:
I agree with @StrikerRUS , I think that's simpler than having this flag and it's ok. We install `pandas` in CI so the tests won't be skipped there.

Reply:
But how would I define the series within the parametrize arguments? I can maybe make a second test that runs for series only and skip that one if pandas isn't installed. Or maybe just skip the `test_list_to_1d_numpy` altogether and just test the training with the column array?

Reply:
I believe you could use `pd_Series` class from our `compat.py` module.
https://github.com/microsoft/LightGBM/blob/4ae59494ab9091a8405a61ae7b13bdedbdbed93e/python-package/lightgbm/compat.py#L4-L17

Will this work?
```py
@pytest.mark.skipif(not PANDAS_INSTALLED, reason='pandas is not installed') 
@pytest.mark.parametrize(
    'y',
    [
        (np.random.rand(10)),
        (np.random.rand(10, 1)),
        ([1] * 10),
        (pd_Series(np.random.rand(10))),
        (pd_Series(['a', 'b'])),
        ([[1], [2]])
    ])
@pytest.mark.parametrize('dtype', [np.float32, np.float64])
def test_list_to_1d_numpy(y, dtype):
```

Reply:
I hadn't seen that, that definitely works, thank you!